src/storable.rs

```diff
@@ -297,6 +298,39 @@ impl<const N: usize> BoundedStorable for [u8; N] {
     const IS_FIXED_SIZE: bool = true;
 }
 
+impl Storable for Principal {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(self.as_slice().to_vec())
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        Self::from_slice(bytes.as_ref())
+    }
+}
+
+impl BoundedStorable for Principal {
+    const MAX_SIZE: u32 = 30;
```

Comment:
The max size of a principal is 29 bytes.

Reply:
it's 29 bytes, plus 1 byte for the length.